xlators/protocol/server/src/server.c

```diff
@@ -825,6 +826,22 @@ server_reconfigure(xlator_t *this, dict_t *options)
         kid = this;
     }
 
+    if (dict_get_int32_sizen(options, "inode-table-size", &inode_table_size) ==
+        0) {
+        conf->inode_table_size = inode_table_size;
+        gf_msg_trace(this->name, 0,
+                     "Reconfigured inode-table-size to "
+                     "%d",
+                     conf->inode_table_size);
+
+        /* traverse through the xlator graph. For each xlator in the
+           graph check whether it is a bound_xl or not (bound_xl means
+           the xlator will have its itable pointer set). If so, then
+           set the lru limit for the itable.
+        */
+        xlator_foreach(this, xlator_set_inode_table_size, &inode_table_size);
```

Comment:
I don't think you need to call xlator_foreach here to configure the inode_table_size, you can pass the value(inode_table_size, dentry_hash_size)during call inode_table_new in the function server_setvolume.

Reply:
I was following the example from network.inode-lru-limit, will make the required update here, thanks for pointing this out.  So just save this into the conf variable then.